specs/chain/hs/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -158,11 +158,13 @@ instance HasTrace CHAIN where
           , _dSEnvLiveness = us ^. dLiveness }
     dCerts <- dcertsGen dsEnv
     let bh
-          = BlockHeader
-          { _prevHHash = h
-          , _bSlot = Slot (s + slotInc)
-          , _bIssuer = vkI
-          , _bSig = Sig vkI (owner vkI)
+          = MkBlockHeader
+          { _bhPrevHash = h
+          , _bhHash = undefined
```

Comment:
Should actually initialise this to something - we should be able to generate hashes easily enough.

Reply:
> Should actually initialise this to something - we should be able to generate hashes easily enough.

I'm trying to see from the LaTeX specification what `bhHash` would be. Should this actually be a function of the type `:: BlockHeader -> Hash`, and not a field of a `BlockHeader`?

Where can I see how to compute this hash?

Reply:
Yes, indeed, it should be. Sorry I didn't spot that.

Reply:
> > Should actually initialise this to something - we should be able to generate hashes easily enough.
> 
> I'm trying to see from the LaTeX specification what `bhHash` should be. Should this actually be a function of the type `:: BlockHeader -> Hash`, and not a field of a `BlockHeader`?
> 
> Where can I see how to compute this hash?

@nc6 or anyone else, can you help me here?

Reply:
What is this hash type here anyway? Does the ledger specify a specific hash?

Reply:
> What is this hash type here anyway? Does the ledger specify a specific hash?

Yes, at the moment we have:

```haskell
-- | An encoded hash of part of the system.
type Hash = Digest SHA256
```

> Where can I see how to compute this hash?

I thought you had defined such a function already:

```haskell
-- | Computes the hash of a header.
hashHeader :: BlockHeader -> Hash
hashHeader = hashlazy . pack . show
-- TODO: we might want to serialize this properly, without using show...
```

Isn't this what we need there?

Reply:
Yes, it specifies a specific hash and in particular this one: https://github.com/input-output-hk/fm-ledger-rules/blob/master/specs/ledger/hs/src/Ledger/Signatures.hs#L10.

Reply:
Thank you, @dnadales ! I fixed this. I take it to be OK now.